cmake/schemes/url_sha1_boost_ios_library.cmake.in

```diff
@@ -157,6 +161,56 @@ hunter_boost_component_b2_args(
 
 list(APPEND build_opts ${b2_component_opts})
 
+#
+# handle iostreams dependency on zlib and bzip2 from
+# within Hunter
+# TODO: DRY
+#
+string(COMPARE EQUAL "@HUNTER_PACKAGE_COMPONENT@" "iostreams" is_iostreams)
+if(is_iostreams)
+  # by default, use both bzip2 and zlib
+  set(IOSTREAMS_NO_COMPRESSION 0)
+  set(IOSTREAMS_NO_ZLIB  0)
+  set(IOSTREAMS_NO_BZIP2 0)
+  set(IOSTREAMS_ZLIB_SOURCE  ) # empty
+  set(IOSTREAMS_BZIP2_SOURCE ) # empty
+  foreach(arg ${b2_component_opts})
+    string(COMPARE EQUAL "-s" "${arg}" is_s)
+    if(NOT is_s)
+      string(REPLACE "=" ";" nvPair ${arg})
+      list(GET nvPair 0 n)
+      list(GET nvPair 1 v)
+      set("IOSTREAMS_${n}" ${v})
+    endif()
+  endforeach()
+  if(NOT IOSTREAMS_NO_COMPRESSION)
+    #
+    # install ZLIB and BZIP2 using hunter iff NO_<LIB> is not set to 1 and <lib>_SOURCE has not been defined
+    # and not both of <lib>_INCLUDE and <lib>_LIBPATH are defined.
+    #
+    if((NOT IOSTREAMS_NO_ZLIB)
+        AND (NOT IOSTREAMS_ZLIB_SOURCE)
```

Comment:
May be we need to make a check that this variable is not defined by the user at all. Since it doesn't make sense to do it? User allowed to set: NO_ZLIB, NO_BZIP2, NO_COMPRESSION. Other options must be empty and can be modified only by Hunter.